Makefile

```diff
@@ -54,23 +54,19 @@ ref.o     := $(ref.c:%.c=$(OBJDIR)/%.o)
 occa.c    := $(sort $(wildcard backends/occa/*.c))
 occa.o    := $(occa.c:%.c=$(OBJDIR)/%.o)
 
-# Output color rules
-COLOR_OFFSET = 3
-COLOR = $(shell echo $(rule_path)|cksum|cut -b1-2)
-rule_path = $(notdir $(patsubst %/,%,$(dir $<)))
-rule_file = $(basename $(notdir $@))
-rule_dumb = @echo -e $(rule_path)/$(rule_file)
-rule_term = @echo -e \\e[38\;5\;$(shell echo $(COLOR)+$(COLOR_OFFSET)|bc -l)\;1m\
-             $(rule_path)\\033[m/\\033[\m$(rule_file)\\033[m
-# if TERM=dumb, use it, otherwise switch to the term one
-output = $(if $(TERM:dumb=),$(rule_term),$(rule_dumb))
-
-V ?= 0
-ifeq ($(V),0)
-  quiet = @printf "  %10s %s\n" "$1" "$@"; $($(1))
-else
-  quiet = $($(1))
-endif
+# Output using the 216-color rules mode
+rule_file = $(notdir $(1))
+rule_path = $(patsubst %/,%,$(dir $(1)))
+last_path = $(notdir $(patsubst %/,%,$(dir $(1))))
+ansicolor = $(shell echo $(call last_path,$(1)) | cksum | cut -b1-2 | xargs -IS expr 2 \* S + 17)
```

Comment:
It seems to me that we should hash the entire directory name, not just the last part. That is, if we end up with `tests/occa` and `backends/occa`, they probably shouldn't be colored the same. Problem is that the color scheme is rather fragile (it should work with both light and backgrounds) and consequently this change (to hashing `$(dir $(1))`) makes the colors less aesthetic.

Looking at the [color table](https://en.wikipedia.org/wiki/ANSI_escape_code#8-bit), there are a number of colors that are unreadable on a light background and a number of other colors that are unreadable on a dark background. I'll merge what we have now because it looks decent with our current directory names, but it's a maintenance irritation awaiting us in the future.

Reply:
I agree, lets merge it and improve later.